src/Fixer/Phpdoc/NoSuperfluousPhpdocTagsFixer.php

```diff
@@ -545,7 +558,18 @@ private function annotationIsSuperfluous(
             $actualTypes[] = 'null';
         }
 
-        return $annotationTypes === $this->toComparableNames($actualTypes, $currentSymbol, $symbolShortNames);
+        $actualTypes = $this->toComparableNames($actualTypes, $namespace, $currentSymbol, $symbolShortNames);
+
+        if ($annotationTypes === $actualTypes) {
+            return true;
+        }
+
+        // retry comparison with annotation type unioned with null
+        // phpstan implies the null presence from the native type
+        $annotationTypes[] = 'null';
+        $annotationTypes = $this->toComparableNames($annotationTypes, null, null, []);
+
+        return $annotationTypes === $actualTypes;
```

Comment:
Personally I would squash it into, but it's only cosmetics.

```suggestion
        return $actualTypes === $this->toComparableNames(array_merge($annotationTypes, ['null']), null, null, []);
```